ChrisPerez/test/test.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+/*
+need to parse json
+-good json
+-bad json
+*/
+
+const chai = require('chai')
+const expect = chai.expect;
+const fs = require('fs')
+const jsonParser = require(__dirname + '/../lib/jsonParser.js');
+const chaiHTTP = require('chai-http');
+chai.use(chaiHTTP);
+const request = chai.request;
+
+describe('middleware unit tests', ()=>{
+  it('should read good JSON', (done)=>{
+    let req = fs.createReadStream(__dirname + '/good.json');
+    jsonParser(req, null, ()=>{
+      expect(req.body.message).to.eql('test');
+      done();
+    })
+  })
+  it('should read bad JSON and throw an error', ()=>{
+    let req = fs.createReadStream(__dirname + '/bad.json');
+    jsonParser(req, null, (err)=>{
+      expect(req.err.message).to.eql('invalid Jason');
+      expect(req.err.statusCode).to.eql(422);
+      done();
+    })
+  })
+})
+
+describe('middleware integration tests', ()=>{
+  it('should read good JSON', (done)=>{
+    require(__dirname + '/../lib/server');
```

Comment:
This should go in a mocha before and not at this point.  This will guarantee that the server is up when you go to make the request.  I didn't notice this at first because tests were passing, but if you're server was loading a lot of middleware it could take longer and your request would not find the server.  Ideally you would close the server in a mocha after. Not points off because we haven't explicitly pointed this out before.